src/api/config/routes.rb

```diff
@@ -737,6 +738,12 @@ def self.public_or_about_path?(request)
       resource :staged_requests, controller: 'staged_requests', only: [:create, :destroy], constraints: cons
     end
   end
+
+  controller 'staging/excluded_requests' do
+    post 'staging_excluded_requests/:number/:project_name' => :create, constraints: cons
```

Comment:
Why not moving this up inside the staging resources?

```ruby
scope module: 'staging' do
  resources :staging_projects, only: [:show], param: :name do
    resources :excluded_requests, only: [:create, :destroy]
  end
end
```

My two concerns are:
* the structure /:number/:project_name looks odd
* that we sprinkle the routes to several places

Reply:
It didn't exist when I wrote this, I can move it. What other structure you propose? :confused:

Reply:
I also fell over the :number/:project_name, but the more I thought about it, the more it appeared logic to me. So the question is who wins: logic or consistency :)

Because you ignore request X and if there is only one action, the target project is implicit. But if there are multiple target projects, you can filter through this where to ignore it. So required arguments before optional arguments - makes sense, no? :)

The problem is (and this is why it puzzled Chris and me): everywhere else the project is *the* argument.

Reply:
+1 for consistency :smile:

Reply:
> Why not moving this up inside the staging resources?

Excluding request doesn't belong in any way to a staging project.

Reply:
It belongs to the staging scope for sure.

> Excluding request doesn't belong in any way to a staging project. It belongs to a staging workflow

I know, my code was not intended for c&p but for talking about it :smile: I still think that we should use resources and be consistent with the other routes (that was already my opinion a few weeks ago when we had discussion with no conclusion). 

I will approve the PR but I don't like that we have now different routes everywhere at all. I really don't like it.

Reply:
Always logic @coolo :wink:

Reply:
Thanks for not even moving it into the scope :angry: :angry: :angry:

Reply:
Sorry, I merged it after you approved it... why did you approved it? hahaha
Sorry man

Reply:
Read my comment :thinking:

Reply:
> Because you ignore request X and if there is only one action, the target project is implicit. But if there are multiple target projects, you can filter through this where to ignore it. So required arguments before optional arguments - makes sense, no? :)

@coolo This is anyway not supported :smile: Currently the parameter is required and also we don't use the action in the controller to get the target_project ... so not logic :smile:

Reply:
then it sounds like a plan not to require it - then Ana's route at least has a reason